### ISSUE TITLE
When using Qt Creator: cannot find -lGL

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#9300]
Creating and building a default "Qt Widgets Application" results in

```
...(compile output window)
g++ -Wl,-rpath,/nix/store/1w7h7p6s2srfw2ady90k7072991lrnpp-qtbase-5.4.2/lib -o qt-test3 main.o mainwindow.o moc_mainwindow.o   -L/nix/store/1w7h7p6s2srfw2ady90k7072991lrnpp-qtbase-5.4.2/lib -lQt5Widgets -lQt5Gui -lQt5Core -lGL -lpthread 
/nix/store/b8qhjrwf8sf9ggkjxqqav7f1m6w83bh0-binutils-2.23.1/bin/ld: cannot find -lGL
collect2: error: ld returned 1 exit status
```

One solution is to manually add ```LIBS += -L/run/opengl-driver/lib``` to the .pro file. But is it possible to do something with qtcreator / qt so that we don't have to do this manual step?

If I'm not mistaken, I think we want qtcreator to try linking with mesa (not ```/run/opengl-driver/lib```, which is NixOS specific) and then have the built program pick a possibly different openGL provider at runtime.

**Comment by bjornfor** [comment]
CC @bbenoist

**Comment by bjornfor** [comment]
CC @vcunat (for mesa question).

**Comment by vcunat** [comment]
No, `/run/opengl-driver*` typically isn't available during build-time, I believe. You should link to provided libGL (from `mesa_noglu`).

**Comment by bjornfor** [comment]
I wonder if this can be considered an upstream bug, because I think if qmake used ```pkg-config --libs --cflags gl``` it would all work just fine.

**Comment by bjornfor** [comment]
@vcunat: Exactly.

**Comment by bjornfor** [comment]
Maybe we can modify the mkspecs file in qtbase:

```
-QMAKE_LIBS_OPENGL       = -lGL
+QMAKE_LIBS_OPENGL       = -lGL -L${mesa_noglu}/lib
```

Mesa is already in the closure.

**Comment by vcunat** [comment]
Was this fixed by #10103?

**Comment by bjornfor** [comment]
@vcunat: Yes and no :-)

There was one commit before that that did the -L part, and later there was https://github.com/NixOS/nixpkgs/pull/10103 that fixed the -I part (I didn't see the issue with missing -I until later). Both fixes are included in master and release-15.09.

**Comment by vcunat** [comment]
Good :-)
